src/utilstrencodings.cpp

```diff
@@ -446,6 +459,34 @@ bool ParseInt32(const std::string& str, int32_t *out)
         n <= std::numeric_limits<int32_t>::max();
 }
 
+bool ParseInt64(const std::string& str, int64_t *out)
+{
+    if (!ParsePrechecks(str))
+        return false;
+    char *endp = NULL;
+    errno = 0; // strtoll will not set errno if valid
+    long long int n = strtoll(str.c_str(), &endp, 10);
```

Comment:
"In other than the C or POSIX locales, other implementation-defined subject sequences may be accepted."

http://pubs.opengroup.org/onlinepubs/009695399/functions/strtoll.html